src/base/utils/fs.cpp

```diff
@@ -311,20 +311,41 @@ bool Utils::Fs::renameFile(const Path &from, const Path &to)
  *
  * This function will try to fix the file permissions before removing it.
  */
-bool Utils::Fs::removeFile(const Path &path)
+nonstd::expected<void, QString> Utils::Fs::removeFile(const Path &path)
 {
-    if (QFile::remove(path.data()))
-        return true;
+    QFile file {path.data()};
+    if (file.remove())
+        return {};
+
+    if (!file.exists())
+        return {};
```

Comment:
This check look redundant, maybe it can be removed?
Or change it to an assertion: `Q_ASSERT(file.exists());`

The same in `moveFileToTrash()`.

Reply:
> This check look redundant

Why? It is here to prevent returning error if `QFile::remove()` above returned false because of "file not exists".
Anyway this logic was here previously and I don't intend to change it (at least within this PR).

Reply:
> Or change it to an assertion: `Q_ASSERT(file.exists());`

It is a really bad idea to assert on something outside of our own code logic.
You can't even oblige the caller to check for the file exists, because it's pointless.

Reply:
>Why? It is here to prevent returning error if QFile::remove() above returned false because of "file not exists".

OK. I missed that case.